test/apps/iodemo/io_demo.cc

```diff
@@ -1309,22 +1359,17 @@ class DemoClient : public P2pDemoCommon {
             assert(_status == OK);
             assert(server_info.retry_count < opts().retries);
 
-            server_info.conn = connect(opts().servers[server_index]);
-            if (server_info.conn == NULL) {
-                connect_failed(server_index);
-                if (_status != OK) {
-                    break;
-                }
-                continue;
-            }
-
-            server_info.retry_count                = 0;
-            _server_index_lookup[server_info.conn] = server_index;
-            active_servers_add(server_index);
-
-            LOG << "Connected to " << server_name(server_index);
+            connect(server_index);
+            assert(server_info.conn != NULL);
+            assert(_status == OK);
```

Comment:
IMHO there is a point until `Connection` implementation can affect `UcxContext` state
I replaced `if` condition with `assert` because `connect` is changed this way

Reply:
via callback